cms/djangoapps/contentstore/views/item.py

```diff
@@ -556,6 +568,10 @@ def _save_xblock(user, xblock, data=None, children_strings=None, metadata=None,
             for field_name in fields:
                 setattr(xblock, field_name, fields[field_name])
 
+            discussion_enabled = fields.get('discussion_enabled')
+            if isinstance(discussion_enabled, bool) and xblock.location.category in ["sequential", "chapter"]:
+                set_discussion_toggle(discussion_enabled, xblock, user)
+
```

Comment:
```suggestion
            discussion_enabled = fields.pop('discussion_enabled', None)
            if isinstance(discussion_enabled, bool) and xblock.location.block_type in ["vertical", "sequential", "chapter", "course"]:
                set_discussion_toggle(discussion_enabled, xblock, user)
                
            for field_name in fields:
                setattr(xblock, field_name, fields[field_name])            
```

I'm suggesting this change for a couple of reasons. 
1. If you don't pop `discussion_enabled` it will be set on blocks that don't support it as well, via setattr
2. I'm adding vertical and course so that they can also be applied this setting in a consistent manner.
3. Renamed category to block_type since it seems that's the new way to get that.